src/stp/tactics/passive/Formation.cpp

```diff
@@ -20,6 +20,8 @@ Formation::Formation() {
 std::optional<StpInfo> Formation::calculateInfoForSkill(StpInfo const &info) noexcept {
     StpInfo skillStpInfo = info;
 
+    if (!info.getPositionToMoveTo()) return std::nullopt;
+
```

Comment:
Here you first make a copy of StpInfo, and then check if you should return `nullopt`. In that case, the copying would have been redundant. It should be more efficient to first check if there is no position to mote to before copying it and setting the dribbler speed to 0.

Reply:
For consistency with other tactics, and because this most likely has very little impact on performance, it has been decided to not change this